### ISSUE TITLE
Share code between ios CDVWKInAppBrowser and CDVUIInaAppBrowser

### ISSUE THREAD

**Issue by jonathanli2** [apache/cordova-plugin-inappbrowser#339]
Hi,
Currently both CDVWKInAppBrowser and CDVUIInAppBrowser are inherited from generic root interface of CDVPlugin. However, since CDVWKInAppBrowser and CDVUIInAppBrowser have the similar functions, they have many functions with the same implementation. 

In order to avoid the duplicated code for easy maintenance, it is better to have a common base interface (like CDVInAppBrowserBase) containing the common code for both CDVUIInAppBrowser and CDVWKInAppBrowser interface. 

Actually, it makes sense if the logic can be refactored to use CDVInAppBrowser as the base interface, and then derives CDVWKWebView and CDVUIWebView from it, instead of deriving all of them from CDVPlugin.

Thanks
Jonathan

**Comment by dpa99c** [comment]
In principle I agree, and my original intention when creating the WKWebView implementation was to do this. In fact I began by attempting to have a common base class from which the two implementations inherited. However, it quickly became clear to me that class inheritance in Objective C is a messy business and I found I was spending more time on the inheritance issues than the actual new WKWebView implementation, hence I decided for brevity to go with the separate implementations which contain duplicity. Also in the back of my mind is that UIWebView is deprecated and may well be removed in a future iOS release, so the effort of an inheritance strategy may be wasted effort.

Having said that, I'm sure if you were to submit a PR which implemented such an inheritance strategy that it would be accepted and merged.
